evalml/data_checks/target_leakage_data_check.py

```diff
@@ -63,14 +62,17 @@ def validate(self, X, y):
         y = _convert_to_woodwork_structure(y)
         X = _convert_woodwork_types_wrapper(X.to_dataframe())
         y = _convert_woodwork_types_wrapper(y.to_series())
-
-        if y.dtype not in numeric_and_boolean_dtypes:
+        combined = X.copy()
+        combined['target'] = y
```

Comment:
What if the original data had a column named "target"? What would happen here?

Reply:
Ah, good call! Will fix that.